src/main.rs

```diff
@@ -1,46 +1,131 @@
 use anyhow::{Result, Context, anyhow, bail};
-use bitcoin::{PrivateKey, Network};
+use bitcoin::{Amount, PrivateKey, Network};
+use bitcoincore_rpc::Client;
 
-mod config;
-mod db;
+use pico_bitcoin_wallet::{config, db};
 
-fn load_private_key() -> Result<PrivateKey> {
-    let data_dir = dirs::data_dir().ok_or(anyhow!("The user data directory was not identified"))?.join("pico-bitcoin-wallet");
-    std::fs::create_dir_all(&data_dir).with_context(|| format!("Failed to create data dir at {}", data_dir.display()))?;
-    let path = data_dir.join("private.key");
-    match std::fs::read_to_string(&path) {
-        Ok(key) => key.parse().context("Failed to parse private key"),
-        Err(error) if error.kind() == std::io::ErrorKind::NotFound => {
-            let key = PrivateKey::new(secp256k1::SecretKey::new(&mut rand::thread_rng()), Network::Regtest);
-            std::fs::write(&path, key.to_wif().as_bytes()).context("failed to save private key")?;
-            Ok(key)
-        },
-        Err(error) => Err(anyhow!(error).context("failed to read private key")),
+fn main() -> Result<()> {
+    let mut args = std::env::args();
+    args.next().ok_or_else(|| anyhow!("program name missing"))?;
+    match args.next() {
+        None => {
+            println!("Command missing\n\n");
+            Ok(help())
+        }
+        Some(command) => {
+            match &*command {
+                "scan" => scan(),
+                "address" => address(),
+                "balance" => balance(),
+                "send" => send(args),
+                "path" => path(),
+                "help" | "--help" |"-h" => Ok(help()),
+                _ => bail!("Unknown command: `{}`", command),
+            }
+        }
     }
 }
 
+/// Prints an address associated with the private key loaded from file.
+///
+/// In a production wallet one would never reuse a single address like this but for demonstration
+/// purposes it will suffice.
+///
+/// You can use a taproot address if you would like to play with taproot spends or alternatively you
+/// can use a native segwit address. Note that the PSBT signing APIs are slightly different for each.
+fn address() -> Result<()> {
+    todo!("Use one of the lines below")
+    // get_segwit_v0_address()
+    // get_taproot_address()
+}
+
+// fn get_taproot_address() -> Result<Address> {}
+// fn get_segwit_v0_address() -> Result<Address> {}
```

Comment:
IMO this is more complicated than before.

Reply:
Removed leaving just the `address` function.